examples/knative/powershell/kn-ps-webhook/handler.ps1

```diff
@@ -0,0 +1,123 @@
+
+   Write-Host "$(Get-Date) - Shutdown Processing Completed`n"
+}
+
+Function Process-Handler {
+   [CmdletBinding()]
+   param(
+      [Parameter(Position=0,Mandatory=$true)]$Headers,
+      [Parameter(Position=1,Mandatory=$true)]$Context
+   )
+
+   Import-Module CloudEvents.Sdk
+
+   Write-host "$(Get-Date) - Processing Context ...`n"
+   try {
+      if ($Context -is [string]) {
+         $Context = [System.Text.Encoding]::UTF8.GetBytes($Context)
+      }
+   } catch {
+      throw "`nFailed to encode `$Context input"
+   }
+
+   try {
+      $jsonSecrets = ${env:WEBHOOK_SECRET} | ConvertFrom-Json
+   } catch {
+      throw "`nK8s secrets `$env:WEBHOOK_SECRET does not look to be defined"
+   }
+
+   if(${env:FUNCTION_DEBUG} -eq "true") {
+      Write-Host "$(Get-Date) - [Function Secrets]:`n${env:WEBHOOK_SECRET}`n"
+   }
+
+   # Check to see if basic auth has been configured for webhook
+   if(${jsonSecrets}.WEBHOOK_USERNAME -ne $NULL -and ${jsonSecrets}.WEBHOOK_PASSWORD -ne $NULL) {
+      Write-host "$(Get-Date) - Checking for authorization ...`n"
+
+      # Expected base64 value for the configured webhook username/pass
+      $expectedPair = "$(${jsonSecrets}.WEBHOOK_USERNAME):$(${jsonSecrets}.WEBHOOK_PASSWORD)"
+      $expectedBytes = [System.Text.Encoding]::ASCII.GetBytes($expectedPair)
+      $expectedBase64 = "Basic " + [System.Convert]::ToBase64String($expectedBytes)
+
+      if(${env:FUNCTION_DEBUG} -eq "true") {
+         Write-Host "$(Get-Date) - [Expected Authorization Header]:`n$expectedBase64`n"
+      }
+
+      # Check for the supplied webhook username/pass
+      if(${Headers}.Authorization) {
+         if($(${Headers}.Authorization) -ne "$expectedBase64") {
+            Write-Error "$(Get-Date) - Invalid Authorization`n"
+            $exception = New-Object Exception
+            $httpException = New-Object System.Net.Http.HttpRequestException -ArgumentList "Invalid Authorization",$exception,400
+            throw $httpException
+         }
+      } else {
+         Write-Error "$(Get-Date) - Authorization header was not provided`n"
+         $exception = New-Object Exception
+         $httpException = New-Object System.Net.Http.HttpRequestException -ArgumentList "Authorization header was not provided",$exception,400
+         throw $httpException
+      }
+   }
+
+   $webhookBody = [System.Text.Encoding]::UTF8.GetString($Context) | ConvertFrom-Json
+
+   if(${env:FUNCTION_DEBUG} -eq "true") {
+      Write-Host "$(Get-Date) - [Headers]:"
+      Write-Host $($Headers | Out-String)
+      Write-Host
+      Write-Host "$(Get-Date) - [Body]:"
+      Write-Host $($webhookBody)
+      Write-Host
+   }
+
+   Write-host "$(Get-Date) - Creating custom CloudEvent ...`n"
+   try {
+      $cloudEvent = New-CloudEvent -Type 'webhook.vmsnapshot.v0' -Source 'kn-ps-webhook' -Id (New-Guid).Guid -Time (Get-Date)
+      $cloudEvent | Set-CloudEventJsonData -Data @{
+         'UserName' = ${webhookBody}.UserName
```

Comment:
As raised earlier, make this logic generic and don't expect a certain data shape.